bytecomp/translcore.ml

```diff
@@ -464,6 +464,11 @@ let specialize_primitive p env ty ~has_constant_constructor =
     | (Pmakeblock(tag, mut, None), fields) ->
         let shape = List.map (Typeopt.value_kind env) fields in
         Pmakeblock(tag, mut, Some shape)
+    | (Patomic_load _, _) ->
```

Comment:
Looks good below, but this pattern should match `Patomic_load { immediate_or_pointer = Pointer }` rather than `Patomic_load _`: if the atomic load is already immediate, then we don't want to redetect it and risk turning it back into Pointer.

Reply:
All right. That makes sense. I will do that. Thanks you!